corehq/apps/custom_data_fields/models.py

```diff
@@ -10,27 +10,23 @@
 CUSTOM_DATA_FIELD_PREFIX = "data-field"
 # This list is used to grandfather in existing data, any new fields should use
 # the system prefix defined below
-SYSTEM_FIELDS = ("commtrack-supply-point",)
+SYSTEM_FIELDS = ("commtrack-supply-point", 'name', 'type', 'owner_id', 'external_id', 'hq_user_id')
 SYSTEM_PREFIX = "commcare"
-RESERVED_WORDS = ('name', 'type', 'owner_id', 'external_id', 'hq_user_id')
 
 
-def _validate_reserved_words(slug, words=SYSTEM_FIELDS):
-    if slug in words:
-        return _('You may not use "{}" as a field name').format(slug)
+def validate_reserved_words(slug):
+    if slug in SYSTEM_FIELDS:
+        raise ValidationError(_('You may not use "{}" as a field name').format(slug))
     for prefix in [SYSTEM_PREFIX, 'xml']:
         if slug and slug.startswith(prefix):
-            return _('Field names may not begin with "{}"').format(prefix)
+            raise ValidationError(_('Field names may not begin with "{}"').format(prefix))
 
 
 def is_system_key(slug):
-    return bool(_validate_reserved_words(slug))
-
-
-def validate_reserved_words(slug):
-    error = _validate_reserved_words(slug, SYSTEM_FIELDS + RESERVED_WORDS)
-    if error is not None:
-        raise ValidationError(error)
+    try:
+        validate_reserved_words(slug)
+    except ValidationError:
+        return True
```

Comment:
I had it like that, and the only reason I removed it was that I felt annoyed at removing a three-line function only to change this one-line function into a six-line function. :disappointed: This will evaluate to `None`, of course, but can't disagree with *Explicit is better than implicit*. I'll have to ask Guido to make a ternary try statement:
```python
    return False try validate_reserved_words(slug) except True
```